src/util/arith_tools.h

```diff
@@ -55,6 +55,71 @@ struct numeric_castt<mp_integer> final
   }
 };
 
+template <typename Tag>
+struct numeric_castt<named_mp_integert<Tag>> final
+{
+  optionalt<named_mp_integert<Tag>> operator()(const exprt &expr) const
+  {
+    if(expr.id() != ID_constant)
+      return {};
+    else
+      return operator()(to_constant_expr(expr));
+  }
+
+  optionalt<named_mp_integert<Tag>> operator()(const constant_exprt &expr) const
+  {
+    mp_integer out;
+    if(to_integer(expr, out))
+      return {};
+    else
+      return named_mp_integert<Tag>{out};
+  }
+};
+
+#if 0
```

Comment:
`#if 0` ?
If this is not needed, please remove.  If it is there for documentation or aesthetic reasons, please document.

Reply:
Ouch, debugging left-over, now removed. Thank you for carefully reading over this extensive change!